### PR TITLE
Ammonia!

### PR DESCRIPTION
**Pull request by Drew4484**
#### Summary
Content "Ammonia and ammonia solution"

#### Purpose of change
Splits ammonia into ammonia and ammonia solution. The former is an extremely useful compound with many applications in chemistry, fertilizer, fermentation, fuels, and more. The latter is the common household cleaner.

#### Describe the solution
Changes most non-lab spawns of ammonia to ammonia solution. Most recipes now use ammonia instead of solution.

#### Describe alternatives you've considered
Ultimately I want ammonia as a chemical for powering gadgets in Aftershock, but I could find alternatives.

#### Testing
WIP

#### Additional context
I intend to leave mutagens using ammonia instead of changing to use the ammonia solutions. Ammonia is still common, especially in labs, and having a process to craft it, even if 'gated' behind a basecamp, will be useful for things like fertilizer.

#55937 provides a route to create ammonia which I'd like to have merged before this is applied. However that PR is still being worked on, so I'll leave this in Draft for now.

### CODE REVIEW

File: data/json/items/chemicals_and_resources.json
Comment (TheShadowFerret) [inline data/json/items/chemicals_and_resources.json:277]:
```suggestion
    "description": "This is anhydrous ammonia, a useful chemical compound.  It must be stored under pressure or refrigerated.  Don't even think about drinking it.",
```

**Comment by AtomicFox556** [comment]
> Add production route, probably Haber-Bosch using iron as the catalyst. Likely requires a basecamp.

Exactly what I did in #55937. Basecamp isn't necessary to build it, and it is producing liquid ammonia, as my assumption was that produced ammonia gas is dissolved in water. It does require parts of industrial equipment that may take a while to find, however.

**Comment by Drew4484** [comment]
> > Add production route, probably Haber-Bosch using iron as the catalyst. Likely requires a basecamp.
> 
> Exactly what I did in #55937. Basecamp isn't necessary to build it, and it is producing liquid ammonia, as my assumption was that produced ammonia gas is dissolved in water. It does require parts of industrial equipment that may take a while to find, however.

Basecamp makes sense to me for large furniture and power supplies, but primarily because this a a complex process that costs a lot of time. If a solo survivor can build and operate it on their own, how long are they stuck standing there per batch?

**Comment by AtomicFox556** [comment]
> Basecamp makes sense to me for large furniture and power supplies, but primarily because this a a complex process that costs a lot of time. If a solo survivor can build and operate it on their own, how long are they stuck standing there per batch?

30 minutes for each 2 units of ammonia, currently. This is a placeholder right now as I found no info regarding how much time Haber-Bosch process takes if you power on the machinery, produce the amount that you needed, and then turn it off. If you could find something about that, it'd be helpful.

File: data/json/items/chemicals_and_resources.json
Comment (TheShadowFerret) [reply data/json/items/chemicals_and_resources.json:277]: You didn't commit the changes? 🤔

**Comment by Drew4484** [comment]
> > Basecamp makes sense to me for large furniture and power supplies, but primarily because this a a complex process that costs a lot of time. If a solo survivor can build and operate it on their own, how long are they stuck standing there per batch?
> 
> 30 minutes for each 2 units of ammonia, currently. This is a placeholder right now as I found no info regarding how much time Haber-Bosch process takes if you power on the machinery, produce the amount that you needed, and then turn it off. If you could find something about that, it'd be helpful.
> 
> Supposedly it's a continuous process, so one could just figure out amount of NH3 produced per hour and then base crafting time on that, but no information on this either.
> 
> Also it deviates from the standard process by sourcing hydrogen from water electrolysis rather than methane cracking. One could first produce methane for it via Sabatier process, but it'd need yet more machinery and I don't think it'd improve overall power efficiency much, if at all.

I'll leave the production up to you then, and just focus on the rest.

**Comment by AtomicFox556** [comment]
> #55937 provides a route to create ammonia which I'd like to have merged before this is applied. However that PR is still being worked on, so I'll leave this in Draft for now.

Actually it's finished now. I'm just waiting until it's merged as I have already fixed all the issues that people have pointed out.

**Comment by AtomicFox556** [comment]
> #55937 provides a route to create ammonia which I'd like to have merged before this is applied. However that PR is still being worked on, so I'll leave this in Draft for now.

Actually it's finished now. I'm just waiting until it's merged as I have already fixed all the issues that people have pointed out (or someone to point out more issues,that is).

File: data/json/items/fuel.json
Comment (dseguin) [inline data/json/items/fuel.json:271]:
```suggestion
    "material": "liquid_ammonia",
```

**Review by LyleSY — APPROVED** [review]
Changes to DinoMod make sense and appear to be consistent with the rest of the PR

**Pull request merged**
